<a href="https://colab.research.google.com/github/VotelessCows/Fismat/blob/main/M2_1_RicardoVargasGardu%C3%B1o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Actividad M2.1 Uso de Google Colab
## Tecnologico de Monterrey
### Equipo 4



Ricardo Vargas Garduño A01026909

Rogelio Lizárraga Escobar A01742161

Juan Pablo Sada San José A01722098

Cecilia Paláu Dávila A01721673

Jorge Lazarini González A00834023



Cambio 1
Cambio prueba en colab

In [ ]:
import pandas as pd

In [ ]:
#Lectura de datos
from google.colab import drive
drive.mount('/content/drive')

df = pd.read_csv("/content/drive/MyDrive/Ciencia de Datos/titanic.csv",  header=0)

Mounted at /content/drive


In [ ]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

##Descripción de Variables

In [ ]:
categorica = ['PassengerId','Survived','Pclass','Name','Sex', 'Ticket', 'Cabin','Embarked'] #Variables categóricas
numerica = ['Age', 'SibSp','Parch', 'Fare']#Variables numéricas

In [ ]:
df.columns #Te regresa las columnas que conforman tu dataframe y su tipo de dato

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [ ]:
df.info() #Te regresa la información de cada columna de tu dataframe, su nombre, la cantidad de datos no nullos , 
#tipo de dato que almacena cada columna y la memoria que ocupa el Dataframe

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
df.describe() #Te regresa diferentes metricas de los datos almacenados en cada columna, 
#te regresa la cantidad de datos en cada columna, una media de todos los datos de cada columna, 
#la desviación estandar de los datos, el valor mas pequeño que almacena, los cuartiles (25%, 50%, 75$) y el valor más grande en la columna

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [ ]:
TotaldeRegistros = len(df)
registros_nulos = df.isnull().any(axis=1)
registros_sin_nulos = df.dropna()

print("El total de registros es:", TotaldeRegistros)
print()
print()
print()
print()
print("Registros con valores nulos:")
print(df[registros_nulos])
print()
print()
print()
print()
print("Registros sin valores nulos:")
print(registros_sin_nulos)

El total de registros es: 891




Registros con valores nulos:
     PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       

Usando describe(), unicamente aparecen las columnas que almacenen datos de tipo int o float, ya que al usarse estadistica, no se puede trabajar con otro tipo de datos.

La tarfia más cara fue de $512.329200

El promedio de edad es 29.699118	años

In [ ]:
for variable in categorica:
  valores_unicos = df[variable].unique()
  cantidad = len(valores_unicos)
  print(f"La variable '{variable}' tiene {cantidad} valores únicos:")
  print()

La variable 'PassengerId' tiene 891 valores únicos:

La variable 'Survived' tiene 2 valores únicos:

La variable 'Pclass' tiene 3 valores únicos:

La variable 'Name' tiene 891 valores únicos:

La variable 'Sex' tiene 2 valores únicos:

La variable 'Ticket' tiene 681 valores únicos:

La variable 'Cabin' tiene 148 valores únicos:

La variable 'Embarked' tiene 4 valores únicos:



##Análisis de Datos

In [ ]:
Total_tarifa = df['Fare'].sum()
print("La tarifa total fue de: $",Total_tarifa)

La tarifa total fue de: $ 28693.9493


In [ ]:
sobrevivientes = df[df['Survived']==1]
no_sobrevivientes = df[df['Survived']==0]

In [ ]:
print("5 personas que sobrevivieron:")
sobrevivientes.head()

5 personas que sobrevivieron:


PassengerId  Survived  Pclass  \
1            2         1       1   
2            3         1       3   
3            4         1       1   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [ ]:
print("5 personas que no sobrevivieron:")
no_sobrevivientes.head()

5 personas que no sobrevivieron:


PassengerId  Survived  Pclass                            Name   Sex   Age  \
0            1         0       3         Braund, Mr. Owen Harris  male  22.0   
4            5         0       3        Allen, Mr. William Henry  male  35.0   
5            6         0       3                Moran, Mr. James  male   NaN   
6            7         0       1         McCarthy, Mr. Timothy J  male  54.0   
7            8         0       3  Palsson, Master. Gosta Leonard  male   2.0   

   SibSp  Parch     Ticket     Fare Cabin Embarked  
0      1      0  A/5 21171   7.2500   NaN        S  
4      0      0     373450   8.0500   NaN        S  
5      0      0     330877   8.4583   NaN        Q  
6      0      0      17463  51.8625   E46        S  
7      3      1     349909  21.0750   NaN        S